hathor/transaction/storage/cache_storage.py

```diff
@@ -207,6 +207,12 @@ def get_all_transactions(self):
 
     def get_count_tx_blocks(self) -> int:
         self._flush_to_storage(self.dirty_txs.copy())
+
+        if self.store.is_rocksdb_storage():
```

Comment:
I don't understand why you need this here. You are already handling things in the `TransactionRocksDBStorage .get_count_tx_blocks(...)` method.

Anyway, I don't feel the cache storage should change its behavior according to the real storage behind it.

Reply:
Because when using cache, the underlying storage has no indices enabled, only the cache storage. So `TransactionRocksDBStorage` would always fallback to using the RocksDB key estimation if cache is enabled.

I wanted to make sure we would still avoid this even when using cache.

But I think @jansegre 's suggestion in https://github.com/HathorNetwork/hathor-core/pull/496/files#r992298106 is  better way to solve this.

What do you think?

I'll give it a try and see if it works.

Reply:
Thanks!